### DISCUSSION TITLE
Can't add tests when creating an assignment

### DISCUSSION THREAD

**Discussion by soggybag** [community/community#101511] [GitHub Education]
### Hi everyone,

I've created a repo with some problems and a unit test. When I use this to create an assignment, when I get to the "Set up autograding and feedback." the "+ Add Test" button doesn't do anything. What I am missing? 

- I'm using Jest for tests. I took the code from the sample repo. 
- my repo is a template.
- I have a package.json with a test script
- I have a test file named .test.js

**Comment by marknelsonengineer** [comment]
I don't think you're missing anything... GitHub Classroom's Autograder seems to be broken right now.  I reported it 2 weeks ago, but it's still broke.

**Reply by soggybag** [reply]
Just to be clear, I'm having problems just adding a test, it won't let me choose a text file. I can't get past the the last step of creating an assignment.

**Comment by Christine-Pinto** [comment]
Hello!

It sounds like you're encountering a hiccup while setting up autograding for an assignment on GitHub, particularly with adding tests using Jest. Let's troubleshoot this step by step to see what might be causing the issue:

1. **Browser Check**: Sometimes, issues like these can be related to browser glitches. Have you tried using a different browser, or clearing your current browser's cache and cookies? It might sound simple, but it’s often a quick fix for such problems.

2. **GitHub Permissions**: Ensure that you have the necessary permissions to modify the repository settings and set up autograding. If you're not the owner, you might need additional permissions from the repository owner.

3. **GitHub Classroom Setup**: Are you setting up the assignment through GitHub Classroom? If so, make sure that your GitHub Classroom is properly linked to your GitHub account and has the necessary permissions to access your repositories.

4. **File Naming and Structure**: Double-check your file structure and naming conventions. The test file should be named with the `.test.js` suffix and located in a directory that Jest is configured to recognize. Also, ensure that your `package.json` is correctly set up with a test script that points to your Jest tests.

5. **Jest Configuration**: Verify your Jest configuration. If you have a custom configuration set up in your `jest.config.js` or in the `package.json`, make sure it's correctly pointing to your test files.

6. **Repository as a Template**: Since your repository is a template, ensure that all necessary files for the test (including `node_modules` if not in `.gitignore`) are included. Sometimes, certain files might be missing or not properly referenced in the template.

7. **Manual Test Execution**: As a sanity check, try running your Jest tests manually in your local environment to ensure they are working as expected. If they run without issues locally, the problem is more likely related to the GitHub setup rather than the tests themselves.

If none of these steps resolve the issue, it might be helpful to share more specific details or screenshots (if possible) of the problem you're encountering. This could provide more insights into what might be going wrong. 

Best of luck, and I hope you get to the bottom of this soon! Happy coding! 🌟

**Reply by soggybag** [reply]
Thanks for the reply. Looks like this is a problem with Safari. The "+ Add Test" button does not reveal the menu on Safari. Seems like a bug.

**Reply by marknelsonengineer** [reply]
I'm having identical problems on Chrome (on my Mac) as well.
